core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -202,7 +202,9 @@ public int readVInt() throws IOException {
             return i;
         }
         b = readByte();
-        assert (b & 0x80) == 0;
+        if ((b & 0x80) != 0) {
+            throw new IOException("Invalid vInt " + Integer.toHexString(b) + " & 0x80 = " + Integer.toHexString((b & 0x80)));
```

Comment:
`Integer.toHexString(b & 0x80):` Isn't this going to be `0x80` 100% of the time? that we are here? I think maybe we should output the current value of the accumulated bytes though (`i`)?